src/main/java/net/snowflake/client/core/arrow/MapConverter.java

```diff
@@ -21,13 +28,20 @@ public MapConverter(MapVector valueVector, int columnIndex, DataConversionContex
   public Object toObject(int index) throws SFException {
     List<JsonStringHashMap<String, Object>> entriesList =
         (List<JsonStringHashMap<String, Object>>) vector.getObject(index);
-    return entriesList.stream()
-        .collect(
-            Collectors.toMap(entry -> entry.get("key").toString(), entry -> entry.get("value")));
+    return StructuredTypeConversionHelper.mapHashMapToObject(entriesList);
   }
 
   @Override
   public String toString(int index) throws SFException {
-    return vector.getObject(index).toString();
+    try {
+      return objectMapper.writeValueAsString(toObject(index));
```

Comment:
Maybe we can have some helper here? The same code is repeated everywhere.

Reply:
There is method in SnowflakeUtil -> SnowflakeUtil.mapJson()

Reply:
Added helper